### PR TITLE
Allow simplified register usage in most cases

### PR DESCRIPTION
**Pull request by lugray**
Allows simplifying:

```ruby
register :Help, 'help', 'dex/commands/help'
register :TwoWords, 'two-words', 'dex/commands/two_words'
```

as

```ruby
self.command_path = "dex/commands"

register :Help
register :TwoWords
```

### CODE REVIEW

File: lib/cli/kit/command_registry.rb [hunk 1]
```diff
@@ -74,6 +82,20 @@ def command_names
       def exist?(name)
         !resolve_command(name).first.nil?
       end
+
+      private
+
+      def snake_case(camel_case, seperator = "_")
+        camel_case.to_s
```
Comment (jules2689) [inline lib/cli/kit/command_registry.rb]: We've tended towards leaving a command providing an example transformation, e.g.

```
camel_case.to_s                                         # MyCoolThing::MyCoolModule
.gsub(/::/, '/')                                        # MyCoolThing/MyCoolModule
.gsub(/([A-Z]+)([A-Z][a-z])/, "\\1#{seperator}\\2")     # I'm not sure what's happening here actually
.gsub(/([a-z\d])([A-Z])/, "\\1#{seperator}\\2")         # I'm not sure what's happening here actually
.downcase
```

Comment (lugray) [reply lib/cli/kit/command_registry.rb]: Updated.

**Comment by burke** [comment]
I'm conflicted: I actually like the explicitness here, that you can see very clearly the command name and file path. Having it inferred from the command name feels magical to me in a way that I'm not completely comfortable with. @jules2689, any opinion on this?

**Comment by lugray** [comment]
I get annoyed by the typing or copy/pasting with small modifications that I have to do whenever I interact with this: adding something, or renaming something.  We always follow the same pattern, and I think we should encourage following the same pattern.

**Comment by burke** [comment]
The main thing we lose here is enabling the reader to understand the mapping to command name and path-on-disk without having to infer based on knowledge of the framework internals.

This does feel like an innocuous preference for writability over readability. What I I don't particularly love is that, in order to fully understand the relationship between two related parts of application code now involves a round-trip through framework idioms.

I guess this underlies my general idea of what makes a good vs. a bad framework: I think that, in general, a good framework should live at the edges, not the heart, of an application, at least to the extent that application code relationships are involved.

On the other hand, the whole `CommandRegistry` paradigm and in particular the `EntryPoint` inheritance API do a completely abysmal job of upholding this ideal, so ¯\\\_(ツ)\_/¯

**Comment by jules2689** [comment]
I think the difference here is the "Rails Way" and a more explicit, less magic way.

There's always parts of rails that I tend to like, but the magic implicitness is oftentimes confusing.
I think I'd prefer the annoyance when changing or adding (which isn't often) and keeping the explicitness? Though, I'm not strongly opinionated on that.

**Pull request closed without merging**